src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -105,19 +100,38 @@ impl FlowServiceImpl {
     async fn run_current_timeslot(
         &self,
         timeslot_time: DateTime<Utc>,
+        catalog_with_transaction: &Catalog,
```

Comment:
This name inspires to suggest another composition-based type for clarity.
How about:

```
struct TransactionalCatalog {
    catalog: Catalog,
}

impl TransactionalCatalog {
    #[inline]
    pub get_catalog(&self) -> &Catalog { &self.catalog }
    
    #[inline]
    pub get_transaction(&self) -> ...
}

```

Reply:
Right now we have to guess what makes this catalog special, and you tried to do this with a name, but I think we should have a more explicit modeling.